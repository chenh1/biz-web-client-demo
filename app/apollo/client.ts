// import { ApolloLink, HttpLink } from "@apollo/client";
// import { onError } from "@apollo/client/link/error";

// import {
//   NextSSRInMemoryCache,
//   NextSSRApolloClient,
// } from "@apollo/experimental-nextjs-app-support/ssr";
// import { NextSSRInMemoryCache } from "@apollo/experimental-nextjs-app-support/ssr";
// import { registerApolloClient } from "@apollo/experimental-nextjs-app-support";}

// const errorLink = onError(({ graphQLErrors, networkError }) => {
//   if (graphQLErrors)
//     graphQLErrors.forEach(({ message, locations, path, ...rest }) => {
//       Object.keys(rest)?.map(key => console.log(key, rest[key]))
//       console.log(
//         `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
//       )
//     });
//   if (networkError) console.log(`[Network error]: ${networkError}`);
// });

// const httpLink = new HttpLink({
//   uri: `${process.env.API_URL || "http://127.0.0.1:1337"}/graphql`,
// })

// export const { getClient } = registerApolloClient(() => {
//   return new NextSSRApolloClient({
//     cache: new NextSSRInMemoryCache(),
//     link: ApolloLink.from([errorLink, httpLink]),
//   });
// });

import { HttpLink } from "@apollo/client";
import {
  registerApolloClient,
  ApolloClient,
  InMemoryCache,
} from "@apollo/experimental-nextjs-app-support";

export const { getClient, query, PreloadQuery } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      // this needs to be an absolute url, as relative urls cannot be used in SSR
      uri: `${process.env.API_URL || "http://127.0.0.1:1337"}/graphql`,
      // you can disable result caching here if you want to
      // (this does not work if you are rendering your page with `export const dynamic = "force-static"`)
      // fetchOptions: { cache: "no-store" },
    }),
  });
});