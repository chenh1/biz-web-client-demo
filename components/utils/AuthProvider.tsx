'use client'

import React from 'react';
import { SessionProvider } from 'next-auth/react'

const AuthProvider: React.FC<{ children?: string | JSX.Element | JSX.Element[] | React.ReactNode }> = ({ children }) => {
    return (
        <SessionProvider>
            {children}
        </SessionProvider>
    )
}

export default AuthProvider