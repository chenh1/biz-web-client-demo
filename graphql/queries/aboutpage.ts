import { gql, DocumentNode } from "@apollo/client";

const aboutPageQuery: DocumentNode = gql`
  query AboutPage {
    aboutPage {
      data {
        attributes {
          title
          content
          topImage {
            data {
              attributes {
                url
                width
                height
                alternativeText
              }
            }
          }
        }
      }
    }
  }
`;

export default aboutPageQuery;