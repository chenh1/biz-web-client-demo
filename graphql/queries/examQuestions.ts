import { gql, DocumentNode } from "@apollo/client";

export type MultiPoints = {
  open: number;
  high: number;
  low: number;
  close: number;
}

export type ChartData = {
  multiPoints: MultiPoints;
  x: number;
  y: number;
}

export type Answer = {
  key: string;
  answer: string;
  value?: string;
  xCell?: number;
  yCell?: number;
  x?: number;
  y?: number;
}

export type Cell = {
  header: string;
  position: number;
}

export type CellValue = {
  value: string | number;
  xCell: number;
  yCell: number;
}

export type Question = {
  question: string;
  correctAnswers: Answer[];
  
}

export type InteractiveTableQuestion = Question & {
  answers: Answer[];
  tableData: CellValue[];
  rowHeaders: Cell[];
  columnHeaders: Cell[];
  answerType: string;
}

export type MultiChoiceQuestion = Question & {
  answers: Answer[];
}

export type InteractiveChartQuestion = Question & {
  xType: string;
  yType: string;
  xAxisLabel: string;
  yAxisLabel: string;
  chartType: "line" | "candlestick";
  chartData: ChartData[];
  interactiveChartData: ChartData[];
  correctAnswer: Answer;
}

export type ExamQuestionType = {
  explanation: string;
  lessonModule: number;
  multiChoiceQuestion?: MultiChoiceQuestion;
  fillInQuestion?: Question;
  interactiveTableQuestion?: InteractiveTableQuestion;
  interactiveChartQuestion?: InteractiveChartQuestion;
  tags: {
    data: {
      attributes: {
        displayName: string;
        color: string;
      }
    }
  }
}

export type ExamQuestionResponse = {
  examQuestions: {
    data: {
      attributes: ExamQuestionType
    }[]
  }
}

export type Datum = {
  [key: string]: number | string;
}

export const examQuestionsQuery: DocumentNode = gql`
  query ExamQuestions($lessonModule: Float) {
    examQuestions(filters: {lessonModule: {eq: $lessonModule }}) {
      data {
        attributes {
          tags {
            data {
              attributes {
                displayName
                color
              }
            }
          }
          explanation
          lessonModule
          multiChoiceQuestion {
            question
            answers {
              answer
              key
            }
            correctAnswers {
              key
            }
          }
          fillInQuestion {
            question
            correctAnswers {
              answer
            }
          }
          interactiveTableQuestion {
            question
            answerType
            tableData(pagination: { page: 1, pageSize: 100 }) {
              value
              xCell
              yCell
            }
            correctAnswers {
              value
              xCell
              yCell
            }
            rowHeaders {
              header
              position
            }
            columnHeaders {
              header
              position
            }
          }
          interactiveChartQuestion {
            xType
            yType
            xAxisLabel
            yAxisLabel
            question
            chartType
            chartData(pagination: { page: 1, pageSize: 100 }) {
              x
              y
              multiPoints
            }
            correctAnswer {
              x
              y
            }
          }
        }
      }
    }
  }
`