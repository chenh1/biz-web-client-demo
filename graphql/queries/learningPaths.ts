import { gql, DocumentNode } from "@apollo/client";

export type RawTree = {
  name: string;
  isCompleted: boolean;
  children?: RawTree[]
}

export const learningPathsQuery: DocumentNode = gql`
  query LearningPaths($level: Int) {
    learningPaths(sort: "level:asc", filters: {level: {eq: $level }}) {
      data {
        attributes {
          title
          level
          description
          rawTree
        }
      }
    }
  }
`

export const allLearningPathsQuery: DocumentNode = gql`
  query LearningPaths {
    learningPaths(sort: "level:asc") {
      data {
        attributes {
          title
          level
          description
          rawTree
        }
      }
    }
  }
`