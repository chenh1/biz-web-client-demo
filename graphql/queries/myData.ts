import { gql, DocumentNode } from "@apollo/client";

export type ProgressData = {
  currentModule?: {
    lessonModule?: string;
  };
  completedModules?: Array<{
    lessonModule: string;
    examScore?: number;
  }>;
}

export type MyDataRes = {
  myData: {
    progressData: ProgressData
  } | {};
}

export const myDataQuery: DocumentNode = gql`
  query MyData {
    myData {
      progressData
    }
  }
`;
