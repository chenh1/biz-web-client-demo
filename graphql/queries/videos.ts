import { gql, DocumentNode } from "@apollo/client";

export type VideoType = {
  title: string;
  url: string;
  description: string;
  transcript: string;
  lessonModule: number | string;
  tags: {
    data: {
      attributes: {
        displayName: string;
        color: string;
        value: string;
      }
    }[]
  }
}

export type Video = {
  attributes: VideoType;
}

export type VideoTypeResponse = {
  videos: {
    data: {
      attributes: VideoType
    }[]
  }
}

export const videosQuery: DocumentNode = gql`
  query Videos($firstModule: Float, $lastModule: Float) {
    videos(pagination: { page: 1, pageSize: 100 }, sort: "lessonModule:asc", filters: {lessonModule: {between: [ $firstModule, $lastModule ] }}) {
      data {
        attributes {
          title
          url
          description
          transcript
          lessonModule
          tags {
            data {
              attributes {
                displayName
                value
                color
              }
            }
          }
        }
      }
    }
  }
`