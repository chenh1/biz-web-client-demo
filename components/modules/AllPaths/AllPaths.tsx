'use client'

import type { ProgressData } from "@/graphql/queries/myData";
import React, { useState, FC } from "react";
import { useQuery } from "@apollo/client";
import { Section } from "../../containers/Section/Section";
import type { SessionWithJwt } from "@/components/constants/types";
import { useSession } from "next-auth/react";
import { useRouter } from "next/navigation";
import { myDataQuery } from "../../../graphql/queries/myData";
import { videosQuery } from "../../../graphql/queries/videos";
import { TreeProgress } from "../TreeProgress/TreeProgress";
import { Heading } from "../../core/Heading/Heading";
import { Box } from "../../containers/Box/Box";
import { nightBlack, hyperRed, forceYellow, ultraPurple, cyberTeal, apexBlue } from '../../constants/colors';
import { colorClasses } from "../../constants/colors";
import { Paragraph } from "../../core/Paragraph/Paragraph";
import { Badge } from "../Deck/ExamView/Badge";
import getScoreRank from "../../utils/getScoreRank";
import { allLearningPathsQuery } from "../../../graphql/queries/learningPaths";
import { QueryResult } from "@apollo/client";

const nodeColors = [hyperRed, forceYellow, ultraPurple, cyberTeal, apexBlue];

interface VideoAttributes {
  lessonModule: string;
  title: string;
  description: string;
}

interface Video {
  attributes: VideoAttributes;
}

interface LearningPathAttributes {
  rawTree: any;
  title: string;
  description: string;
  level: string;
}

interface LearningPath {
  attributes: LearningPathAttributes;
}

interface AllPathsQueryData {
  learningPaths: {
    data: LearningPath[];
  };
}

interface VideosQueryData {
  videos: {
    data: Video[];
  };
}

interface MyDataQueryData {
  myData: {
    progressData?: ProgressData;
  };
}

export const AllPaths: FC = () => {
  const session = useSession().data as SessionWithJwt | null;
  const router = useRouter();

  const { data: allLearningPathsData } = useQuery<AllPathsQueryData>(allLearningPathsQuery);
  const { data: videosData, loading, error } = useQuery<VideosQueryData>(videosQuery, { variables: { firstModule: 0, lastModule: 999 } });
  const { data: myData } = useQuery<MyDataQueryData>(myDataQuery, {
    context: {
      headers: {
        'Authorization': `Bearer ${session?.data?.jwt}`
      },
    },
    skip: session?.status !== 'authenticated'
  });

  const allLearningPaths = allLearningPathsData?.learningPaths.data || [];
  const videos = videosData?.videos.data || [];
  const progressData = myData?.myData.progressData || {};

  // const currentModule = videos.find(({ attributes: { lessonModule } }) => lessonModule === progressData?.currentModule?.lessonModule)?.attributes || {};
  const [nodeData, setNodeData] = useState<any[]>([]);
  
  return (
    <>
      {allLearningPaths.map(({ attributes: { rawTree, title, description, level } }, i) => {
        const targetNodeData = nodeData[i];
        const previewModule: VideoAttributes | undefined = videos.find(({ attributes: { lessonModule } }) => lessonModule === targetNodeData?.data.name)?.attributes!;
        const examScore = progressData?.completedModules?.find(({ lessonModule }) => lessonModule === targetNodeData?.data.name)?.examScore;
        const tree = JSON.parse(JSON.stringify(rawTree));
        const nodeCount = tree?.nodeCount;
        const rank = getScoreRank(examScore ?? 0);
        const scoreMessage = `You scored ${Math.round((examScore ?? 0) * 100)}%`
        
        return (
          <Section key={i}>
            <Box padding="0" gap="lg" className="sm:grid-cols-3">
              <Box padding="0" className="relative col-span-1 self-center">
                <div className={`absolute opacity-25 left-0 h-36 w-36 grid place-items-center rounded-full bg-${colorClasses[i]}`}>
                  <Heading type="h4" fontSizeClass="text-8xl">{level}</Heading>
                </div>
                <Box padding="0" className="z-10">
                  <Heading type="h3">{title}</Heading>
                  <Paragraph>{description}</Paragraph>
                </Box>
              </Box>
              {tree &&
                <div style={{ minHeight: `${nodeCount * 100}px` }}>
                  <TreeProgress
                    isVertical
                    hideRootNode
                    globalOnMouseEnter={(node) => {
                      const newNodeData = [...(nodeData || [])];
                      newNodeData[i] = node;
                      setNodeData(newNodeData);
                    }}
                    parentNodeStroke={nodeColors[i]}
                    progressData={progressData}
                    rawTree={tree}
                    highlightValue={parseInt(progressData?.currentModule?.lessonModule ?? "")}
                    className="col-span-1 w-full h-full"
                    background={nightBlack}
                  />
                </div>
              }
              <Box padding="0" gap="sm" className="self-center">
                {previewModule &&
                  <>
                    <div className="flex">
                      <div className="CodeMirror-merge-2pane">
                        <Heading type="h3">{previewModule?.title}</Heading>
                      </div>
                      {!!examScore &&
                        <div className="relative">
                          <div className="absolute left-0 bottom-0 scale-[.25] -mx-6 -my-16">
                            <Badge shouldLimitOverflow={false} rank={rank} />
                          </div>
                        </div>
                      }
                    </div>
                    <Paragraph>{previewModule?.description}</Paragraph>
                    {!!examScore && <Paragraph className="italic" textColor="hyper-red" darkTextColor="hyper-red">{scoreMessage}</Paragraph>}
                  </>
                }
              </Box>
            </Box>
          </Section>
        );
      })}
    </>
  );
}
