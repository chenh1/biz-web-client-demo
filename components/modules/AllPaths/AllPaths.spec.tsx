import { AllPaths } from "./AllPaths";
import { MockedProvider } from "@apollo/client/testing";
import { render, waitFor } from '@testing-library/react';
import { JSDOM } from "jsdom"
import { allLearningPathsQuery } from "../../../graphql/queries/learningPaths";
import { videosQuery } from "../../../graphql/queries/videos";
import { myDataQuery } from "../../../graphql/queries/myData";
import AuthProvider from "../../utils/AuthProvider";
const dom = new JSDOM()
global.document = dom.window.document
global.window = dom.window

jest.mock("next/navigation", () => ({
  useRouter() {
    return {
      prefetch: () => null
    };
  },
  useParams() {
    return {
      id: "1"
    }
  }
}));

jest.mock("@visx/responsive", () => ({
  ParentSize: ({ children }: { children: (size: { width: number, height: number }) => JSX.Element }) => children({ width: 100, height: 100 })
}))

// mock LocalResizeObserver
global.LocalResizeObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}))


const mocks = [
  {
    request: {
      query: allLearningPathsQuery,
    },
    result: {
      data: {
        learningPaths: {
          data: [
            { attributes: { title: "Intro to cooking", level: "1", description: "Learn the basics of cooking", rawTree: JSON.parse(JSON.stringify({ name: "Lesson 1", description: "Some description", isCompleted: false, children: [ { name: "Lesson 2", description: "Some description", isCompleted: false, children: [] } ] })) } },
            { attributes: { title: "Intermediate cooking", level: "2", description: "Take the next steps", rawTree: JSON.parse(JSON.stringify({ name: "Lesson 1", description: "Some description", isCompleted: false, children: [ { name: "Lesson 2", description: "Some description", isCompleted: false, children: [] } ] })) } },
            { attributes: { title: "Master chef", level: "3", description: "Gain master in cooking", rawTree: JSON.parse(JSON.stringify({ name: "Lesson 1", description: "Some description", isCompleted: false, children: [ { name: "Lesson 2", description: "Some description", isCompleted: false, children: [] } ] })) } }
          ]
        }
      }
    }
  },
  {
    request: {
      query: videosQuery,
      variables: { firstModule: 0, lastModule: 999 }
    },
    result: {
      data: {
        videos: {
          data: [
            { attributes: { title: "Video 1", url: "www.supercoolmastercheftime.com", description: "Video 1", transcript: "Some words", lessonModule: 1.01 }},
            { attributes: { title: "Video 2", url: "www.supercoolmastercheftime.com", description: "Video 2", transcript: "Some words again", lessonModule: 1.02 }}
          ]
        }
      }
    }
  },
  {
    request: {
      query: myDataQuery,
    },
    result: {
      data: {
        myData: {
          progressData: {
            currentModule: {
              lessonModule: 1.01
            },
            completedModules: [
              { lessonModule: 1.01, examScore: 100 },
              { lessonModule: 1.02, examScore: 100 }
            ]
          }
        }
      }
    }
  },
];


it('renders multiple series of lessons in a vertical node format', async () => {
  const { container } = render(
    <MockedProvider mocks={mocks}>
      <AuthProvider>
        <AllPaths />
      </AuthProvider>
    </MockedProvider>
  );

  await waitFor(() => new Promise((res) => setTimeout(res, 800)));

  expect(container).toMatchSnapshot();
});
