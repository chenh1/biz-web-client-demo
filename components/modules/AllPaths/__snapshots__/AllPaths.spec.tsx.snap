// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders multiple series of lessons in a vertical node format 1`] = `
<div>
  <div
    class=" flex border-box justify-around w-full py-24 "
  >
    <div
      class="w-full max-w-7xl px-4 xl:px-0"
    >
      <div
        class=" grid  gap-12  p-2 sm:grid-cols-3"
      >
        <div
          class=" grid  gap-4  p-2 relative col-span-1 self-center"
        >
          <div
            class="absolute opacity-25 left-0 h-36 w-36 grid place-items-center rounded-full bg-hyper-red"
          >
            <h4
              class="text-night-black dark:text-max-white font-headings font-semibold text-8xl "
            >
              1
            </h4>
          </div>
          <div
            class=" grid  gap-4  p-2 z-10"
          >
            <h3
              class="text-night-black dark:text-max-white font-headings font-semibold text-2xl sm:text-4xl "
            >
              Intro to cooking
            </h3>
            <p
              class=" text-night-black dark:text-max-white font-body font-body text-base undefined"
            >
              Learn the basics of cooking
            </p>
          </div>
        </div>
        <div
          style="min-height: NaNpx;"
        >
          <div
            class=" grid  gap-4  p-2 overflow-y-auto col-span-1 w-full h-full"
          >
            <svg
              height="100"
              width="100"
            >
              <defs>
                <lineargradient
                  id="lg"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="1"
                >
                  <stop
                    offset="0%"
                    stop-color="#fd9b93"
                    stop-opacity="1"
                  />
                  <stop
                    offset="100%"
                    stop-color="#fe6e9e"
                    stop-opacity="1"
                  />
                </lineargradient>
              </defs>
              <rect
                fill="#050E1C"
                height="100"
                rx="14"
                width="100"
              />
              <g
                class="visx-group"
                transform="translate(80, 80)"
              >
                <g
                  class="visx-group cursor-pointer relative"
                  transform="translate(-30, 0)"
                >
                  <circle
                    data-cy="-node-0"
                    fill="#050E1C"
                    height="56"
                    r="28"
                    stroke="#FF2F97"
                    stroke-width="1"
                    width="56"
                    x="-28"
                    y="-28"
                  />
                  <text
                    class=""
                    data-cy="-node-text-0"
                    dx="0"
                    dy=".33em"
                    fill="#ffffff"
                    font-family=""Space Grotesk", sans-serif"
                    font-size="18"
                    style="pointer-events: none;"
                    text-anchor="middle"
                  >
                    Lesson 2
                  </text>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <div
          class=" grid  gap-4  p-2 self-center"
        />
      </div>
    </div>
  </div>
  <div
    class=" flex border-box justify-around w-full py-24 "
  >
    <div
      class="w-full max-w-7xl px-4 xl:px-0"
    >
      <div
        class=" grid  gap-12  p-2 sm:grid-cols-3"
      >
        <div
          class=" grid  gap-4  p-2 relative col-span-1 self-center"
        >
          <div
            class="absolute opacity-25 left-0 h-36 w-36 grid place-items-center rounded-full bg-force-yellow"
          >
            <h4
              class="text-night-black dark:text-max-white font-headings font-semibold text-8xl "
            >
              2
            </h4>
          </div>
          <div
            class=" grid  gap-4  p-2 z-10"
          >
            <h3
              class="text-night-black dark:text-max-white font-headings font-semibold text-2xl sm:text-4xl "
            >
              Intermediate cooking
            </h3>
            <p
              class=" text-night-black dark:text-max-white font-body font-body text-base undefined"
            >
              Take the next steps
            </p>
          </div>
        </div>
        <div
          style="min-height: NaNpx;"
        >
          <div
            class=" grid  gap-4  p-2 overflow-y-auto col-span-1 w-full h-full"
          >
            <svg
              height="100"
              width="100"
            >
              <defs>
                <lineargradient
                  id="lg"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="1"
                >
                  <stop
                    offset="0%"
                    stop-color="#fd9b93"
                    stop-opacity="1"
                  />
                  <stop
                    offset="100%"
                    stop-color="#fe6e9e"
                    stop-opacity="1"
                  />
                </lineargradient>
              </defs>
              <rect
                fill="#050E1C"
                height="100"
                rx="14"
                width="100"
              />
              <g
                class="visx-group"
                transform="translate(80, 80)"
              >
                <g
                  class="visx-group cursor-pointer relative"
                  transform="translate(-30, 0)"
                >
                  <circle
                    data-cy="-node-0"
                    fill="#050E1C"
                    height="56"
                    r="28"
                    stroke="#FDD701"
                    stroke-width="1"
                    width="56"
                    x="-28"
                    y="-28"
                  />
                  <text
                    class=""
                    data-cy="-node-text-0"
                    dx="0"
                    dy=".33em"
                    fill="#ffffff"
                    font-family=""Space Grotesk", sans-serif"
                    font-size="18"
                    style="pointer-events: none;"
                    text-anchor="middle"
                  >
                    Lesson 2
                  </text>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <div
          class=" grid  gap-4  p-2 self-center"
        />
      </div>
    </div>
  </div>
  <div
    class=" flex border-box justify-around w-full py-24 "
  >
    <div
      class="w-full max-w-7xl px-4 xl:px-0"
    >
      <div
        class=" grid  gap-12  p-2 sm:grid-cols-3"
      >
        <div
          class=" grid  gap-4  p-2 relative col-span-1 self-center"
        >
          <div
            class="absolute opacity-25 left-0 h-36 w-36 grid place-items-center rounded-full bg-ultra-purple"
          >
            <h4
              class="text-night-black dark:text-max-white font-headings font-semibold text-8xl "
            >
              3
            </h4>
          </div>
          <div
            class=" grid  gap-4  p-2 z-10"
          >
            <h3
              class="text-night-black dark:text-max-white font-headings font-semibold text-2xl sm:text-4xl "
            >
              Master chef
            </h3>
            <p
              class=" text-night-black dark:text-max-white font-body font-body text-base undefined"
            >
              Gain master in cooking
            </p>
          </div>
        </div>
        <div
          style="min-height: NaNpx;"
        >
          <div
            class=" grid  gap-4  p-2 overflow-y-auto col-span-1 w-full h-full"
          >
            <svg
              height="100"
              width="100"
            >
              <defs>
                <lineargradient
                  id="lg"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="1"
                >
                  <stop
                    offset="0%"
                    stop-color="#fd9b93"
                    stop-opacity="1"
                  />
                  <stop
                    offset="100%"
                    stop-color="#fe6e9e"
                    stop-opacity="1"
                  />
                </lineargradient>
              </defs>
              <rect
                fill="#050E1C"
                height="100"
                rx="14"
                width="100"
              />
              <g
                class="visx-group"
                transform="translate(80, 80)"
              >
                <g
                  class="visx-group cursor-pointer relative"
                  transform="translate(-30, 0)"
                >
                  <circle
                    data-cy="-node-0"
                    fill="#050E1C"
                    height="56"
                    r="28"
                    stroke="#FF31F3"
                    stroke-width="1"
                    width="56"
                    x="-28"
                    y="-28"
                  />
                  <text
                    class=""
                    data-cy="-node-text-0"
                    dx="0"
                    dy=".33em"
                    fill="#ffffff"
                    font-family=""Space Grotesk", sans-serif"
                    font-size="18"
                    style="pointer-events: none;"
                    text-anchor="middle"
                  >
                    Lesson 2
                  </text>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <div
          class=" grid  gap-4  p-2 self-center"
        />
      </div>
    </div>
  </div>
</div>
`;
